### PR TITLE
Fix CI SDK distribution configuration

### PR DESCRIPTION
**Pull request by Guardiola31337**
- This PR fixes CI SDK distribution configuration

cc @zugaldia @LukasPaczos

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -267,13 +313,61 @@ jobs:
...
+      #     - run:
+      #         name: Check public documentation
+      #         command: sh scripts/dokka-validate.sh
+      - deploy:
+          name: Publish Navigation SDK 1.0 to Bintray
+          command: |
+            version=$(cat gradle/artifact-settings.gradle | grep "versionName")
+            if [[ $version != *"SNAPSHOT"* ]]; then
+              make 1.0-publish-to-bintray
+            fi
```
Comment (LukasPaczos) [inline circle.yml]: This whole flow can be extracted to a single job that accepts the Bintray repo argument.

**Comment by Guardiola31337** [comment]
🆗 Addressed feedback. Ready for another round of 👀

File: circle.yml [hunk 2]
```diff
@@ -97,48 +293,20 @@ jobs:
...
-          command: make 1.0-build-debug
-      - run:
-          name: Run unit-test
-          command: make 1.0-unit-tests
+      - restore-gradle-cache-1_0
+      - static-code-analysis-kotlin
+      - build-debug-1_0
+      - unit-tests-1_0
 
   ui-tests:
```
Comment (LukasPaczos) [inline circle.yml]: Should we call this `ui-tests-1_0`?

**Comment by LukasPaczos** [comment]
Actually, we can compare the old process on CI history. The legacy `build` took ~9:20 and `release` took ~3:50 but they run in parallel, so the whole execution took ~9:20. Now, the consolidated build takes ~10:30.

Not a big deal, but if the builds start growing, it might make sense to split them up.

**Comment by Guardiola31337** [comment]
@LukasPaczos 
> One downside of the PR is that we'll build both release and debug builds for both legacy and 1.0 in one CI job respectively. This might increase the time of this job while otherwise, they would run in parallel.

> Actually, we can compare the old process on CI history. The legacy `build` took ~9:20 and `release` took ~3:50 but they run in parallel, so the whole execution took ~9:20. Now, the consolidated build takes ~10:30.

Nah that was the case before too 👀 https://github.com/mapbox/mapbox-navigation-android/pull/2498/files#diff-29944324a3cbf9f4bd0162dfe3975d88L58

> Not a big deal, but if the builds start growing, it might make sense to split them up.

We can split it and add it as required in the future to scratch some time 👍

**Pull request merged**